src/cascadia/TerminalControl/ControlCore.cpp

```diff
@@ -2245,27 +2248,20 @@ namespace winrt::Microsoft::Terminal::Control::implementation
 
     Windows::Foundation::Collections::IVector<Control::ScrollMark> ControlCore::ScrollMarks() const
     {
-        auto internalMarks{ _terminal->GetScrollMarks() };
-        auto v = winrt::single_threaded_observable_vector<Control::ScrollMark>();
-        for (const auto& mark : internalMarks)
-        {
-            Control::ScrollMark m{};
+        const auto& internalMarks = _terminal->GetScrollMarks();
+        std::vector<Control::ScrollMark> v;
 
-            // sneaky: always evaluate the color of the mark to a real value
-            // before shoving it into the optional. If the mark doesn't have a
-            // specific color set, we'll use the value from the color table
-            // that's appropriate for this category of mark. If we do have a
-            // color set, then great we'll use that. The TermControl can then
-            // always use the value in the Mark regardless if it was actually
-            // set or not.
-            m.Color = OptionalFromColor(_terminal->GetColorForMark(mark));
-            m.Start = mark.start.to_core_point();
-            m.End = mark.end.to_core_point();
+        v.reserve(internalMarks.size());
 
-            v.Append(m);
+        for (const auto& mark : internalMarks)
+        {
+            v.emplace_back(
+                mark.start.to_core_point(),
+                mark.end.to_core_point(),
+                OptionalFromColor(_terminal->GetColorForMark(mark)));
         }
 
-        return v;
+        return winrt::single_threaded_vector(std::move(v));

```

Comment:
Adding this to the PR was a bit of a mistake (I did `git restore` from another branch). But it might be something we'd want to keep anyways - up to you all! 😅
The benefits are:
* Much much faster vector creation (without WinRT indirection)
* Reserve the right amount of memory
* `emplace_back` constructs the trivial struct in-place instead of constructing it on the stack first and then copying it over to the heap

BTW I think this method is missing lock guards. Scroll marks are being written by the VT thread after all. I'm sure my other PR adds them. ^^

Reply:
absolutely fine with me